rerun_cpp/src/rerun/status.hpp

```diff
@@ -9,23 +9,28 @@
 
 struct rr_status;
 
+namespace arrow {
+    class Status;
+}
+
 namespace rerun {
     /// Status codes returned by the SDK as part of `Status`.
     ///
     /// Category codes are used to group errors together, but are never returned directly.
     enum class StatusCode : uint32_t {
         Ok = 0,
+        OutOfMemory = 1,
 
         // Invalid argument errors.
-        _CategoryArgument = 0x000000010,
+        _CategoryArgument = 0x00000010,
```

Comment:
this can help readability:

```suggestion
        _CategoryArgument = 0x0000'0010,
```

Reply:
oh nice didn't know that's a thing. I tried 0x0000_000 and it didn't like that